src/actions/firestore.js

```diff
@@ -188,76 +188,87 @@ export function setListener(firebase, dispatch, queryOpts, successCb, errorCb) {
   const meta = getQueryConfig(queryOpts);
 
   // Create listener
-  const unsubscribe = firestoreRef(firebase, meta).onSnapshot(
```

Comment:
None of the internals change in the onSnapshot. I just move it so uses could optionally add an `includeMetadataChanges` to get listeners to optionally trigger when Firestore writes data to the local IndexDB. This is important when in Li-Fi or offline mode so optimistic data in the cache reducer stays to the minimum.